docs/contents/usage.md

```diff
@@ -0,0 +1,84 @@
+# Usage
+
+## Setup
+
+First, `dmenv` needs a Python3 interpreter in PATH, which should be called `python` or `python3`. This should already be the case if you've just installed Python3, regardless of your operating system.
+
+Second, `dmenv` needs a `setup.py` file to work.
+
+* If you don't have a `setup.py` yet, you can run `dmenv init <project name>`
+  to generate one. In this case, make sure to read the comments inside
+  and edit it to fit your needs.
+
+* If you already have one, please note that `dmenv` uses the `extras_require` keyword with a `dev` key
+  to specify development dependencies, which you can use to replace your `dev-requirements.txt`
+  file for instance.
+
+And that's it. Now you are ready to use `dmenv`!
+
+Here's a description of the main commands:
+
+
+## dmenv lock
+
+Here's what `dmenv lock` does:
+
+* First, it creates a virtualenv for you with `python -m venv` in
+  `.venv/<version>`, where `<version>` is read from `python --version`. Make
+  sure to add `.venv` to your `.gitignore`! Note that this step is skipped
+  if `dmenv` detects it is run from an existing virtualenv.
+
+* Then it runs `pip intall --editable .[dev]` so that your dev deps are installed, and the scripts listed in `entry_points` are
+  created.
+
+* Finally, it runs `pip freeze` to generate a `requirements.lock` file.
+
+Now you can add the `requirements.lock` file to your version control system.
+
+This leads us to the next command.
+
+## dmenv install
+
+Now that the complete list of dependencies and their versions is written in the
+`requirements.lock` file, anyone can run `dmenv install` to install all the
+dependencies and get exactly the same versions you got when you ran `dmenv lock`.
+
+Hooray reproducible builds!
+
+## dmenv run
+
+As a convenience, you can use:`dmenv run` to run any binary from the virtualenv. If the program you want to run
+needs command-line options, use a `--` to separated them from `dmenv` options, like so:
+
+```console
+dmenv run -- pytest --collect-only
```

Comment:
`--` does not seperate the command-line options of the binary but the binary and its command-line options... This does not match the explanation above.